examples/README.md

```diff
@@ -0,0 +1,371 @@
+```
+
+**Quick Start (Knative):**
+```bash
+kubectl apply -f services/greeter/k8s/knative-v1.yaml
+```
+
+**Documentation:**
+- [services/greeter/README.md](services/greeter/README.md) - Service overview and quick reference
+- [services/greeter/TESTING.md](services/greeter/TESTING.md) - Step-by-step test scenarios
+
+## Testing Scenarios
+
+The examples support comprehensive testing of:
+
+### ReplicaSet Deployment
+1. **Basic Deployment** - Deploy v1, verify registration
+2. **Poison Pattern** - Test that v1 fails on "poison" input (no ANTIDOTE)
+3. **Graceful Draining** - Long-running requests during upgrades
+4. **Versioned Upgrade** - Deploy v2 with ANTIDOTE fix (new deployment ID)
+
+### Knative Deployment
+
+#### In-Place Updates (Explicit Tags)
+1. Deploy with tag "v1"
+2. Apply fix (same tag, new Revision)
+3. Verify same deployment ID
+
+#### Versioned Updates (Explicit Tags)
+1. Deploy with tag "v1"
+2. Change tag to "v2"
+3. Verify new Configuration created
+4. Verify multiple Configurations coexist
+
+#### Auto-Versioning (Template Hash)
+1. Deploy without tag
+2. Change template (image, env vars)
+3. Verify new Configuration per template change
+4. Verify automatic versioning
+
+## Architecture Overview
+
+### ReplicaSet Mode
+
+```
+RestateDeployment (greeter)
+    ↓
+ReplicaSet
+    ↓
+Pods
+    ↓
+Service
+    ↓
+Restate Admin API (registration)
+```
+
+**Characteristics:**
+- Manual scaling via `replicas` field
+- Standard Kubernetes pod lifecycle
+- Rolling updates via maxUnavailable/maxSurge
+- No scale-to-zero
+
+### Knative Mode
+
+```
+RestateDeployment (greeter-knative)
+    ↓
+Knative Configuration (greeter-knative-v1)
+    ↓
+Knative Revisions (00001, 00002, ...)
+    ↓
+Knative Route (greeter-knative-v1)
+    ↓
+Restate Admin API (registration)
+```
+
+**Characteristics:**
+- Autoscaling via minScale/maxScale/target
+- Scale-to-zero capability
+- Tag-based deployment identity
+- Gradual rollout (Knative feature)
+- Multiple concurrent deployments (versioned updates)
+
+## Prerequisites
+
+### Required Components
+
+1. **Kubernetes Cluster**
+   - Minikube, kind, or any Kubernetes cluster
+   - Version 1.24 or later recommended
+
+2. **Restate Operator**
+   - Install via Helm or apply manifests directly
+   - See main repository README for installation instructions
+
+3. **For Knative Mode:**
+   - Knative Serving installed
+   - Version 1.12 or later recommended
+   - See https://knative.dev/docs/install/
+
+### Optional Tools
+
+- `kubectl` - Required for all operations
+- `restate` - Restate CLI for managing deployments and invocations
+- `curl` - For testing service endpoints
+- `jq` - For parsing JSON responses
+- `watch` - For monitoring resource changes
+
+### Installing the Restate CLI
+
+```bash
+# Install Restate CLI (see https://docs.restate.dev/operate/cli)
+curl https://restate.dev/install.sh | bash
+
+# Configure CLI to use port-forwarded Restate
+export RESTATE_HOST=localhost:9070
```

Comment:
should be the default unless you have another config set, but you can always do restate -e local or `RESTATE_ENVIRONMENT=local`